tests/test_fix_course_delete_module_class_outcome_test.php

```diff
@@ -0,0 +1,169 @@
+ */
+class test_fix_course_delete_module_class_outcome_test extends \advanced_testcase {
+
+    /**
+     * Test deletion of data after test.
+     *
+     * @coversNothing
+     */
+    public function test_deleting() {
+        global $DB;
+        $this->resetAfterTest(true);
+        $DB->delete_records('course');
+        $DB->delete_records('course_modules');
+        $DB->delete_records('context');
+        $DB->delete_records('assign');
+        $DB->delete_records('quiz');
+        $this->assertEmpty($DB->get_records('course'));
+        $this->assertEmpty($DB->get_records('course_modules'));
+        $this->assertEmpty($DB->get_records('context'));
+        $this->assertEmpty($DB->get_records('assign'));
+        $this->assertEmpty($DB->get_records('quiz'));
+    }
+
+    /**
+     * Test for get/set modulename & get/set contextid.
+     *
+     * @covers \tool_fix_course_delete_module\outcome
+     */
+    public function test_outcome_class() {
+        global $DB;
+        $this->resetAfterTest(true);
+
+        // Setup a course with a page, an assignment and a quiz module.
+        $user     = $this->getDataGenerator()->create_user();
+        $course   = $this->getDataGenerator()->create_course();
+        $page     = $this->getDataGenerator()->create_module('page', array('course' => $course->id));
+        $pagecm   = get_coursemodule_from_id('page', $page->cmid);
+        $assign   = $this->getDataGenerator()->create_module('assign', array('course' => $course->id));
+        $assigncm = get_coursemodule_from_id('assign', $assign->cmid);
+        $quiz     = $this->getDataGenerator()->create_module('quiz', array('course' => $course->id));
+        $quizcm   = get_coursemodule_from_id('quiz', $quiz->cmid);
+
+        // The module exists in the course.
+        $coursedmodules = get_course_mods($course->id);
+        $this->assertCount(3, $coursedmodules);
+
+        // Delete page & quiz table record to replicate failed course_module_delete adhoc tasks.
+        $this->assertCount(1, $DB->get_records('page'));
+        $DB->delete_records('page');
+        $this->assertEmpty($DB->get_records('page'));
+
+        $this->assertCount(1, $DB->get_records('quiz'));
+        $DB->delete_records('quiz');
+        $this->assertEmpty($DB->get_records('quiz'));
+
+        // Setup 1st adhoc task for page module deletion.
+        $removaltaskpage = new \core_course\task\course_delete_modules();
+        $pagedata = [
+            'cms' => [$pagecm],
+            'userid' => $user->id,
+            'realuserid' => $user->id
+        ];
+        $removaltaskpage->set_custom_data($pagedata);
+        \core\task\manager::queue_adhoc_task($removaltaskpage);
+
+        // Setup 2nd adhoc task for a multi-module delete (both quiz and assign).
+        $removaltaskmulti = new \core_course\task\course_delete_modules();
+        $cmsarray = array(''.$assigncm->id => array('id' => $assigncm->id),
```

Comment:
Do you use `''.` to convert int to string?  Would `(string) $assigncm->id` work?

Reply:
fixing